crates/wasmparser/src/validator/operators.rs

```diff
@@ -2048,6 +2048,12 @@ where
         self.push_operand(ty)?;
         Ok(())
     }
+    fn visit_typed_select_multi(&mut self, mut tys: Vec<ValType>) -> Self::Output {
+        if tys.len() != 1 {
+            bail!(self.offset, "invalid result arity");
+        }
+        self.visit_typed_select(tys.pop().unwrap())
```

Comment:
Could this `debug_assert!` that `tys.len() != 1` and then always return with `bail!`?

Reply:
Sure! Done.